src/libguac/guacamole/wol-constants.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef GUAC_WOL_CONSTANTS_H
+#define GUAC_WOL_CONSTANTS_H
+
+/**
+ * Header file that provides constants and defaults related to libguac
+ * Wake-on-LAN support.
+ * 
+ * @file wol-constants.h
+ */
+
+/**
+ * The value for the local IPv4 broadcast address.
+ */
+#define GUAC_WOL_LOCAL_IPV4_BROADCAST "255.255.255.255"
+
+/**
+ * The number of seconds to wait after sending the Wake-on-LAN packet
+ * for the destination host to start responding.
+ */
+#define GUAC_WOL_BOOT_WAIT_TIME 60
```

Comment:
I'm still not sure about this.

The other constants make total sense; `GUAC_WOL_LOCAL_IPV4_BROADCAST` is exactly that, as are `GUAC_WOL_PACKET_SIZE` and `GUAC_WOL_PORT`. Each is a standard value which any WoL implementation should reasonably provide as a reusable constant.

`GUAC_WOL_BOOT_WAIT_TIME` isn't a standard value nor a standard part of WoL. My understanding from reading the code thus far is that this is just a constant which we are using internally to avoid having to define such a constant within each protocol. I agree that we should aim to be DRY and that defining such a constant somewhere makes sense ... but I don't think the public API of libguac is the place for it. I think it's outside the scope of concerns for the core, public library of guacamole-server.

There is a place where we have been placing internally-reusable code that isn't part of the public API (`src/common/`). If it goes anywhere, I think that would be the place for it, even it it's just something like `src/common/common/defaults.h`. Such a header might even serve as a place to receive other default values that have otherwise been duplicated across each protocol implementation but _could_ be common.

If you think this constant does belong in the public libguac API, I think I can be swayed, but I'd like to read through the reasoning. As it stands, I currently can't see it as something within scope.

Reply:
Makes sense.  I've created the `common/defaults.h` file and added it, there, instead.